amtl/os/am-system-errors.h

```diff
@@ -44,7 +44,7 @@ static inline void
 FormatSystemErrorCode(int code, char* error, size_t maxlength) {
 #if defined(KE_WINDOWS)
     if (!FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, nullptr, code,
-                        MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), error, maxlength, nullptr)) {
+                        MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), error, (int)maxlength, nullptr)) {
```

Comment:
If we're doing this, please add:

    assert(maxlength <= std::numeric_limits<DWORD>::max());